lib/puppet/provider/group/groupadd.rb

```diff
@@ -32,13 +32,11 @@ def findgroup(key, value)
     group_file = "/etc/group"
     group_keys = [:group_name, :password, :gid, :user_list]
     index = group_keys.index(key)
-    File.open(group_file) do |f|
-      f.each_line do |line|
-         group = line.split(":")
-         if group[index] == value
-             f.close
-             return Hash[group_keys.zip(group)]
-         end
+    @group_content = File.read(group_file) unless @group_content
```

Comment:
Is there a scenario where this file is modified but we still have the contents in memory?

Reply:
AFAIK no. From my tests it looks like `finduser` is always called before any changes are made to the resource/file:

![image](https://user-images.githubusercontent.com/25789827/70326906-4f8e8280-183e-11ea-8662-2870c6027571.png)

Reply:
Also `finduser` is only called from property getters (`gid`, `comment`...) so the files shouldn't change between property reads.